core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeReservedValueManager.kt

```diff
@@ -311,7 +308,7 @@ class TrackedEntityAttributeReservedValueManager @Inject internal constructor(
         if (remainingValues < minNumberToTryFill) {
             val numberToReserve = fillUpTo!! - remainingValues
             downloadValues(
-                attribute, organisationUnit, numberToReserve, pattern, storeError
```

Comment:
The purpose of this storeError was to optionally store the error based on the method that originated the query: if the query is generated as a side effect of `getValue()`, it is fine if the query fails because the device can be offline and the user didn't trigger the query. If the query is generated explicitly by the user, it is important to store the error.

The `storeError` property can be added to the TrackedEntittyAttributeReservedValueQuery and then applied in the coroutineExecutor.